weed/storage/needle.go

```diff
@@ -55,8 +60,17 @@ func (n *Needle) String() (str string) {
 }
 
 func ParseUpload(r *http.Request) (
-	fileName string, data []byte, mimeType string, isGzipped bool,
+	fileName string, data []byte, mimeType string, pairs []byte, isGzipped bool,
 	modifiedTime uint64, ttl *TTL, isChunkedFile bool, e error) {
+	pairMap := make(map[string]string)
+	for k, v := range r.Header {
+		if len(v) > 0 && strings.HasPrefix(k, PairNamePrefix) {
+			pairMap[k] = v[0]
```

Comment:
@chrislusf Considering the situation when uploading file directly with addition pairs, keys are only trimmed off prefix when needle is stored (in function NewNeedle).